src/rnp/rnpcfg.cpp

```diff
@@ -247,6 +247,10 @@ rnp_cfg_setstr(rnp_cfg_t *cfg, const char *key, const char *val)
         return false;
     }
 
+    std::string mkey("\x01");
+    mkey.append(key);
+    rnp_cfg_addstr(cfg, mkey.c_str(), val? val : "");
+
```

Comment:
You don't need to modify function `rnp_cfg_setstr()` as required functionality is already implemented in `rnp_cfg_addstr()`. Please see how handled `OPT_USERID` and `CFG_SIGNERS` in `rnp.cpp`.